### PR TITLE
Network checker: Initial commit of Functions scenarios

### PR DESCRIPTION
**Pull request by sidkri**
This PR adds Azure Functions specific scenarios to the Network checker diagnostics tool.  It also refactors some code that was shared from another flow by moving the functions to the flowMisc.js file.  I will update the PR with more details.

### CODE REVIEW

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/flowMisc.js [hunk 1]
```diff
@@ -75,3 +74,1042 @@ export async function GetSubnet(armService, subnetResourceId) {
     }
     return subnetData;
 }
+
+export async function RunKuduAccessibleCheck(diagProvider) {
```
Comment (yifguoMSFT) [inline AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/flowMisc.js:78]: nitpick, let's use JavaScript/TypeScript naming convention like this
image

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-checks.component.ts [hunk 2]
```diff
@@ -96,7 +97,7 @@ export class NetworkCheckComponent implements OnInit, AfterViewInit {
             var globals = this._globals;
             globals.messagesData.currentNetworkCheckFlow = null;
             var telemetryService = this._telemetryService;
-            var networkCheckFlows = { connectionFailureFlow, configFailureFlow, learnMoreFlow }
+            var networkCheckFlows = { connectionFailureFlow, configFailureFlow, functionsFlow, learnMoreFlow }
```
Comment (yifguoMSFT) [inline AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-checks.component.ts:100]: We may need to make sure the app is a Functions App then enable this flow

Comment (sidkri) [reply AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-checks.component.ts:100]: Good catch!

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/flowMisc.js (hunk 1 above)
Comment (sidkri) [reply AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/flowMisc.js]: Changed all functions to follow the naming convention

File: AngularApp/package.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
     "applens": "node --max_old_space_size=8192 node_modules/@angular/cli/bin/ng serve applens --aot --optimization=false",
     "ncloud-applens-prod": "ng serve applens --configuration=nationalcloud-production --aot --optimization=false",
     "ncloud-applens-local": "ng serve applens --configuration=nationalcloud-local --aot --optimization=false",
-    "ssl": "node --max_old_space_size=8192 node_modules/@angular/cli/bin/ng serve app-service-diagnostics --aot --optimization=false --ssl --sslCert ./ssl/server.crt --sslKey ./ssl/server.key",
+    "ssl": "node --max_old_space_size=8192 node_modules/@angular/cli/bin/ng serve --host 0.0.0.0 app-service-diagnostics --aot --optimization=false --ssl --sslCert ./ssl/server.crt --sslKey ./ssl/server.key",
```
Comment (sidkri) [inline AngularApp/package.json:11]: This is needed when running within a docker container

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-checks.component.ts (hunk 2 above)
Comment (sidkri) [reply AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-checks.component.ts]: Added conditional logic.

**Comment by sidkri** [comment]
Closed this PR as I will submit another PR that leverages the new UX elements @yifguoMSFT created [#1159]

**Pull request closed without merging**